docker/install/ubuntu_install_caffe.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# Prerequisite
+apt-get update --fix-missing
+
+export DEBIAN_FRONTEND=noninteractive
+apt-get install -y tzdata
+ln -fs /usr/share/zoneinfo/Asia/Shanghai /etc/localtime
+dpkg-reconfigure --frontend noninteractive tzdata
+ 
+apt-get install libprotobuf-dev -y
+apt-get install libleveldb-dev -y
+apt-get install libsnappy-dev -y
+apt-get install libopencv-dev -y
+apt-get install libhdf5-serial-dev -y
+apt-get install protobuf-compiler -y
+apt-get install libgflags-dev -y
+apt-get install libgoogle-glog-dev -y
+apt-get install liblmdb-dev -y
+apt-get install libatlas-base-dev -y
+apt-get install --no-install-recommends libboost-all-dev -y
+apt-get install gfortran -y
+
+cd /
+mkdir caffe
+cd caffe
+wget  https://github.com/weiliu89/caffe/archive/ssd.zip -O ssd.zip
```

Comment:
Do we have special reason we use this custom distribution of Caffe instead of BVLC?

Reply:
Yes, our caffe frontend will support for some layers and networks that has not been implemeted in BVLC, such as ProrBox in SSDMobilenet, and this custom distribution can support all tests in https://github.com/apache/incubator-tvm/pull/6000

Reply:
we should have one discussion about it. Because Caffe proto could be customized, we can not support every distribution. The default version we should support is the official distribution of course. But I also know SSD is one important  usage. How about we maintain one Caffe proto  like MXNet (https://github.com/apache/incubator-mxnet/tree/v1.7.x/tools/caffe_converter) and generate caffe_pb2.py? The Caffe proto should be based on BVLC official version but if we add customization we could discuss. Maybe this is one trade off to support important usage which is not included in the BVLC version. @tqchen @siju-samuel do you have better suggestion about it?

Reply:
I agree with you. In fact, we already maintain one caffe proto file internally, which includes various layers from different caffe distribution. If add a new layer, we will add layer definiton in caffe.proto firstly and then generate caffe_pb2.py which will be used by caffe frontend. But, now the problem is that BVLC can not include all the layers that supported by our Caffe frontend, if we add BVLC distribution in CI, the tests case will fail.

For the Caffe frontend pr, I think there are two steps:
1. CI pr: we can use SSD distribution temporarily， it convers as many layers as possible;
2. Caffe Frontend pr: we can offer caffe.proto too if neccessary.

Reply:
If we add BVLC distribution, we could seperate our pr and avoid the ops BVLC distribution doesn't support.

Reply:
Then if we all agree we should maintain our TVM Caffe proto, in our CI, the caffe_pb will be generated by this Caffe proto (like we could support SSD now). Then we could pass the test in your pr.